server/src/test/java/org/elasticsearch/gateway/MetaDataStateFormatTests.java

```diff
@@ -258,7 +261,9 @@ public void testLoadState() throws IOException {
         }
         List<Path> dirList = Arrays.asList(dirs);
         Collections.shuffle(dirList, random());
-        MetaData loadedMetaData = format.loadLatestState(logger, xContentRegistry(), dirList.toArray(new Path[0]));
+        final boolean hasMissingCustoms = randomBoolean();
```

Comment:
Instead of relying on randomness, can we please have explicit tests for each case? There are not that many combinations here.

Reply:
I've split the test into four (see https://github.com/elastic/elasticsearch/pull/48390/commits/4069cbe086ffc95517b25e8afd56a29fcbb16cc5)